Change default graphics compositor settings so UI isn't affected by post processing or shown in editor

JackPilley opened the pull request:
# PR Details

This PR updates the default graphics compositor setup so that UI is in its own render stage that isn't affected by post processing and also isn't shown in the editor view.

## Description

By default there is now a render stage called UserInterface which the UIRenderFeature is assigned to. The UIRenderFeature is also no longer assigned to the Transparent render stage.

The Editor Renderer entry point is now a SceneRenderCollection with one child, a shared forward renderer which doesn't use the UserInterface stage. This stops the UI from being shown in the editor. SceneRenderCollection was chosen in case a user wants the UI to be visible in the editor. They just need to add a SingleStageRenderer set to the UserInterface stage to the collection.

The Game Renderer remains mostly the same, using a Camera Renderer with a SceneRenderCollection as the child. The collection has an added SingleStageRenderer set to the UserInterface stage.

I consider this setup much simpler than the one described [here](https://github.com/herocrab/StrideCleanUI) while achieving the same results.

A small additional change this PR makes is giving proper names to the render features, they were all just test or null before.

There's a risk that this is a breaking change, so I'd like some feedback on that before marking this as ready to merge.

## Related Issue

Addresses https://github.com/stride3d/stride/issues/1154
This setup also avoids the problem from https://github.com/stride3d/stride/issues/1508

## Motivation and Context

See related issues.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## To do
- [x] Update the level 10 archetype file
- [x] Update the level 9 archetype file
- [ ] Update the create default function in the [GraphicsCompositorHelper class](https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Engine/Rendering/Compositing/GraphicsCompositorHelper.cs)

phr00t committed:
VR: First pass as OpenXR support for Stride

phr00t committed:
VR: Add extension to Vulkan

phr00t committed:
OpenXR: builds now, though has trouble finding the Vulkan device...

phr00t committed:
OpenXR: get rid of old code

Aggror committed:
navigation rework

Aggror committed:
Scene improvements

Aggror committed:
Projecting shortened

Aggror committed:
Audio stream

tebjan committed:
[Materials] Fixes render stage evaluation if the transparency of a material changes (#1346)

Eideren committed:
[Input] Fix mouse release for Winforms (#1304)

Eideren committed:
[Math] Optimize Matrix mult and invert (#1301)

Eideren committed:
[Physics] Fix Rigidbody memory leak (#1297)

Eideren committed:
[Input] Improve Input.Keys docs and fix a couple of keymapping in SDL (#1265)

Eideren committed:
[Graphics] Fix and improve Resource.Reload (#1300)

Eideren committed:
[Physics] Static mesh, asset and editor support (#1251)

Aggror committed:
Async api sample

tebjan committed:
[Rendering] Fixes crash when changing DoF quality at runtime.

tebjan committed:
Merge pull request #1351 from vvvv/dev/tebjan/fix-dof-quality-change-master

[Rendering] Fixes crash when changing DoF quality at runtime

tebjan committed:
Adds info for developers to earn money

tebjan committed:
Fixes link

tebjan committed:
Shorter text

Eideren committed:
[Docs] Moved bounty info to wiki page

Aggror committed:
Added extra questions

Eideren committed:
[Github] Feature request, small typo

eltociear committed:
Fix typo in XAudio2.cpp

acutally -> actually

tebjan committed:
Merge pull request #1354 from eltociear/patch-1

Fix typo in XAudio2.cpp

jrinker03 committed:
Fix for issue #1185 - Stride.Launcher - Could not find a part of the path

The launcher executable was moved to a new tools directory as part of the fix for #1139.  Older packages did not contain a tools directory so added code to check for the existence of a target directory before delving into it.

Also tightened up the code in the affected area and removed an unused using statement.

jrinker03 committed:
Fix for Issue #1361 - Replace deprecated WebRequest & WebClient

1) Replaced all insances of WebRequest with HttpClient plus the appropriate Get + Read operations.

2) Replaced the use of WebClient with HttpClient plus Get + stream read.

3) Removed semi-problematic refereces and the SDK specifier as they caused the compiler to generate a number of warnings.   All these are not needed in .NET 6.

jrinker03 committed:
Updates from code review:
1) HttpResponseMessage (return from HttpClient.GetAsync) should be disposed so added using statements as appropriate.  In some cases the code was slightly rearranged for readability with this update.

2) Since the BaseUrl is never actually used elsewhere, removed updating it in ReleaseNotesViewModel.cs

3) Removed unneeded setting of stream pointers to their head immediately after creation since they will be that way already.

4) In NewsPageViewModel.cs, converted the use of MemoryStream to the return type of ReadAsStreamAsync, Stream, since it allows for the same usage in the following code and avoids a potentially risky cast.

Aggror committed:
wip fpc and tpc

Aggror committed:
working tpc, adjusting racast

Aggror committed:
completed third person raycasts

Aggror committed:
Rearrange scenes

Aggror committed:
Tutorial menu wip

manio143 committed:
Merge pull request #1362 from jrinker03/Issue1361

Fix for Issue #1361 - Replace deprecated WebRequest & WebClient

manio143 committed:
Merge pull request #1359 from jrinker03/master

Fix for issue #1185 - Stride.Launcher - Could not find a part of the path

Aggror committed:
utorial menu fixes and improved on various scenes

tebjan committed:
[Rendering] Fixes null exception on first frame. #1343

tebjan committed:
Merge pull request #1376 from stride3d/dev/tebjan/fix-for-pr-1346

[Rendering] Fixes null exception on first frame. #1343

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

tebjan committed:
Merge pull request #1379 from stride3d/all-contributors/add-jrinker03

docs: add jrinker03 as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

tebjan committed:
Merge pull request #1380 from stride3d/all-contributors/add-tristanmcpherson

docs: add tristanmcpherson as a contributor for code

jrinker03 committed:
Fix for #1363 - remove explicit reference to assemblies in .NET SDK to eliminate compiler warning. (#1364)

jrinker03 committed:
Address compiler warnings by explicitly casting values after examining safety or by changing the type used.    Also removed some commented out debugging code.

Aggror committed:
Some folder numbering and code comments

Aggror committed:
Cleanup

jrinker03 committed:
Warning fixes for C# code, part 1
Fixes the following warnings:
- CS0108 'OldCollectionDescriptor.IsCollection(Type)' hides inherited member 'CollectionDescriptor.IsCollection(Type)'. Use the new keyword if hiding was intended.
- CS0162 Unreachable code
- CS0168 The variable 'e' is declared but never used
- CS0169 The field 'foo' is never used
- CS0184 The given expression is never of the provided ('ProjectViewModel') type

Plus some minor using cleanup and making code a tad more consistent.

jrinker03 committed:
Update stride package definitions to use a license expression of 'MIT' rather than the deprecated licenseUrl.

jrinker03 committed:
Update nuspecs to have correct license information

jrinker03 committed:
From PR: Restore CommandList.scissorsDirty and suppress the warning since it's used conditionally.

Aggror committed:
Added intermediate tutorial project to launcher

Aggror committed:
Added start scenes

manio143 committed:
Merge pull request #1387 from jrinker03/DeprecatedLicenseUrl

Update stride package definitions to use a license expression of 'MIT' rather than the deprecated licenseUrl.

manio143 committed:
Merge pull request #1385 from jrinker03/CsWarnings1

Warning fixes for C# code, part 1

manio143 committed:
Merge pull request #1384 from jrinker03/CppCastFixes

Address CPP compiler warnings by explicitly casting values after examining safety or by changing the type used. Also removed some commented out debugging code.

Aggror committed:
raycast compl

Aggror committed:
rootasset

Aggror committed:
Removing spaces and dots to support templates

Aggror committed:
Removed dot in code files , double code folder

Aggror committed:
Tiny improvements

Aggror committed:
clceanup scenes

gregsn committed:
Added abstractions IIntersectableWithRay (implemented by BoundingBox, BoundingSphere and Plane) and IIntersectableWithPlane (implemented by BoundingBox and BoundingSphere) (#1395)

Aggror committed:
Final testing of scenes

Aggror committed:
remove rider settings

Aggror committed:
Applied PR feedback  thanks Eideren and Manio143

Aggror committed:
Some more typo fixes

Aggror committed:
Removed old unused scripts, small tweaks to scenes

Aggror committed:
Remove unused fonts

Aggror committed:
Shortened the teleporter code

Eideren committed:
[Build] Add .vsconfig to check missing requirements in VS (#1314)

Eideren committed:
[Editor] Fix missing editor keyboard input (#1274)

jrinker03 committed:
[VSPlugin] Support VS 2022 + 2019 Extensions in the launcher (#1411)

Aggror committed:
copyright added to files

Aggror committed:
replaced tutorial icons with eiderens version

Aggror committed:
Character orientation usage instead of transform

Aggror committed:
Merge pull request #1401 from Aggror/C#IntermediateTutorial

Tutorial project added: C#intermediate tutorial

jrinker03 committed:
Removing build-time warnings

Continuing to chip away at numerous varieties of build-time warnings and make minor code cleanup changes.

- Add a runtime ID to Stride.Samples.Tests so the compiler generates the same architecture as its dependencies.
- In Stride.Samples.Tests, remove link to AttachedChildProcessJob.cs as its not needed and results in a duplicate symbol warning.
- In AssetPropertGraph, actually use the isDisposed field.
- In several projects, us the released version of System.Collections.Immutable rather than the prerelease.
- In Stride.Core.Design, allow it to see the internal NativeLockFile compiled into Stride.Core.IO and avoid the duplicate symbol warning.
- Correct a few cases of sealed classes having protected members.
- As hygene, move ParticleBufferState to its own .cs
- In ParticleVertexBuilder, remove the unused indexStructSize
- Across multiple test projects, use the release version of Microsoft.NET.Test.Sdk rather than the prerelease version.

jrinker03 committed:
Do not trim on Stride.Launcher publish step (#1418)

manio143 committed:
Merge pull request #1415 from jrinker03/WarningsRound2

Removing build-time warnings

azeno committed:
Fixes "key already exists" exception when preloading library containing characters in upper case (#1423)

Aggror committed:
tutorial improvements

Aggror committed:
Merge remote-tracking branch 'origin/master' into C#IntermediateTutorial

# Conflicts:
#	README.md

Aggror committed:
UI text

arturoc committed:
work in progress to port openxr impl to directx11

Aggror committed:
Tutorial improvements again

Aggror committed:
Undo incorrect readme merge

Aggror committed:
Merge branch 'master' into C#IntermediateTutorial

jrinker03 committed:
Update setup packages to detect and install .NET 6 (#1420)

* Update setup packages to detect and install .NET 6

Also update to latest Advanced Installer.

* Update installer files:
- Since the Launcher package is a self-contained EXE, there is no need for a prereq on .NET 6.
- Updated the Stride prereqs to drop .NET 5 and include .NET 6 and to use Advanced Installer 19.4

Aggror committed:
Merge pull request #1426 from Aggror/C#IntermediateTutorial

C#intermediate tutorial - improvements

arturoc committed:
More changes to openxr, still not working

arturoc committed:
OpenXr: refactored to order methods better

Now working, was actually working before just missing a setting on the
example.

Rendering is still really slow and there should be a way to sync the
device required by openxr and the one used by stride

arturoc committed:
Texture.Direct3D: revert unneeded change

Aggror committed:
async tutorial

azeno committed:
Fixes assembly load exceptions in asset compiler

manio143 committed:
Merge pull request #1434 from stride3d/dev/azeno/assembly-name-comparison

Fixes assembly load exceptions in asset compiler

arturoc committed:
fixes for oculus

openxr struct type wasn't initialized in many cases

arturoc committed:
openxr: support for tracking space

arturoc committed:
openxr: Fixes for vs2019

arturoc committed:
openxr: remove vulkan code

arturoc committed:
openxr: removing unecesary changes

arturoc committed:
openxr: destroy tracking space before creating a new one

arturoc committed:
Logger.Info -> Logger.Debug

Aggror committed:
small fps improvement

Aggror committed:
cleanup tpc

Aggror committed:
removed line by mistake

Aggror committed:
feedback async await

Aggror committed:
Merge pull request #1444 from Aggror/C#IntermediateTutorial

C#intermediate tutorial

arturoc committed:
openxr: remove custom code from game

not needed by merging commit and flush on OpenXrHmd

arturoc committed:
remove unused variable

arturoc committed:
openxr: dispose

arturoc committed:
openxr: Remove mirror texture check from forward renderer

Instead check it on the settings by using a CanMirrorTexture flag that VRDevices can report back if they don't support texture mirroring

Aggror committed:
C# intermediate tutorial fixes: remade audio startscene. remove specular from wood (#1446)

eltociear committed:
Fix typo in AssignmentExpression.cs (#1447)

assigment -> assignment

joreg committed:
Mesh copy constructor was missing BoundingSphere

tebjan committed:
Merge pull request #1450 from vvvv/dev/joreg/mesh-fix-for-master

Mesh copy constructor was missing BoundingSphere

xen2 committed:
[Build] Update build script to use Advanced Installer 19.4

xen2 committed:
[UI] Fix iOS build

xen2 committed:
[Build] Reorganized digital signatures. Temporarily disabled StrideSetup.exe (causing error 500 when signing)

xen2 committed:
[Setup] Disabled AI_EXTERNALUIUNINSTALLERNAME=aiui since it was causing a error 500 when code signing the setup (?!)

xen2 committed:
[Package] Fix upgrade for net5.0 executable projects from Stride 4.0

xen2 committed:
[Build] Prerequisite installer: make sure to download signclient before calling Advanced Installer

xen2 committed:
Revert "[Editor] Fix missing editor keyboard input (#1274)"

This reverts commit 7015814d32f595f47b96b179444b0f400967d3da.

johang88 committed:
#983 Update DefaultValue attribute for IsXYNormal property

tebjan committed:
Merge pull request #1454 from johang88/983_FixReconstructZDefaultValue

#983 Update DefaultValue attribute for IsXYNormal property

xen2 committed:
[Package] Fix upgrade for macOS/Linux project

xen2 committed:
[Build] Sign after building VSIX package (and sign VSIX itself)

xen2 committed:
[Build] Further fix for VSIX signing

xen2 committed:
[Samples] Bump package

tebjan committed:
[Samples] Fix build error of VRSanbox

Just removing a null check, as the type has changed from class to a struct and cannot be null. Fixes #1457

xen2 committed:
[Samples] Bumped samples version

xen2 committed:
[Samples] Update camera before using its ViewMatrix (otherwise it result in NaN being sent to physics engine, resulting in a crash)

ericwj committed:
Get rid of P/Invoke SizeOf.

ericwj committed:
Bring back `int SizeOf<T>(T[])`

manio143 committed:
Merge pull request #1464 from ericwj/MigrateInterop

Get rid of P/Invoke SizeOf.

ericwj committed:
Migrate IL weaving 10-CoreRuntime and Stride

ericwj committed:
Migrate IL weaving Step 2 for all of Stride.sln.

ericwj committed:
Migrate IL weaving Step 2 for all of Stride.sln.

ericwj committed:
Migrate remaining uses of Stride.Core.Interop.

ericwj committed:
Migrate remaining uses of Stride.Core.Interop.

ericwj committed:
Migrate `RemoveInitLocals` to `SkipLocalsInit` and obsolete `RemoveInitLocalsAttribute`.

azeno committed:
Fixes key not found exception when working with multiple devices (#1463)

* Fixes key not found exception when working with multiple devices

See initial bug report here: https://discourse.vvvv.org/t/multiple-multitouch-screens-crashes-on-fast-touches/20683/5

* Fixes typo

ericwj committed:
Make editing projects easier. (#1470)

* Make editing projects easier.

* Make editing projects easier.

manio143 committed:
Merge pull request #1472 from ericwj/MigrateRemoveInitLocals

Migrate `[RemoveInitLocals]` to `[SkipLocalsInit]` and obsolete `RemoveInitLocalsAttribute`

arturoc committed:
Fix wrong formatted curly brace

ericwj committed:
Merge branch 'stride3d:master' into MigrateInterop3

ericwj committed:
Merge branch 'stride3d:master' into MigrateInterop2

ericwj committed:
Delete `CoreUtilities` and most of `StrideCoreExtensions`.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Merge.

ericwj committed:
Fix compiler errors after merge.

ericwj committed:
Fix compiler errors after merge.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Use `is`. Fix `#define CHECKED`.

ericwj committed:
Use `is`. Fix `#define CHECKED`.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Use is. Fix #define CHECKED.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Fix Vulkan build.

ericwj committed:
Review fixed statements for null and access size.

ericwj committed:
Review fixed statements for null and access size.

ericwj committed:
Merge amend

ericwj committed:
Delete unfinished work in progress.

ericwj committed:
Fix confusion of size with alignment.

ericwj committed:
Bring back `void Interop.Pin<T>(T)` until `UpdateEngine` is redesigned.

ericwj committed:
Rigorously check `ParameterCollection.CopyTo<T>` bounds.

ericwj committed:
Fix `Stride.Rendering.ParameterCollection` buffer allocation size off-by-one alignment padding.

ericwj committed:
Fix Android build.

ericwj committed:
Fix Android build.

ericwj committed:
Change the diagnostic code for UpdateEngine unsafe operations.

ericwj committed:
Implement Stream.Read/Write(Span) and Serialize(Span)

ericwj committed:
Implement Stream.Read/Write(Span) and Serialize(Span)

ericwj committed:
Implement Stream.Read/Write(Span) and Serialize(Span)

ericwj committed:
Remove NativeStream, NativeMemoryStream from inheritance graph.
Remove Read/Write(nint, int), also from SerializationStream.
Polyfill with extension methods. Source compatible; binary breaking.

ericwj committed:
Provide obsoletion messages in BinarySerialization.
Proper exceptions in new methods; fix messages in others.

ericwj committed:
Remove duplicate.

ericwj committed:
Delay initializing `out` variables in `GetHeadPoseUnsafe`.

ericwj committed:
Fix asset compiler build.

ericwj committed:
Fix access violation in particle system simulation processor.

ericwj committed:
Correct exception behavior of derived `System.IO.Stream` types.
Remove manually copied documentation comments in favor of `<inheritdoc/>`.

ericwj committed:
Do not use trailing alignment padding for source parameter collection buffer.
Fixes pre-existing buffer overrun and those introduced by assuming trailing alignment padding.

ericwj committed:
Fix whitespace in LZ4Stream.cs

ericwj committed:
Rewrite parameter buffer copying.
To account for differences in type/size of equally named parameters.

ericwj committed:
`StreamExtensions` should have been public.

Basewq committed:
[Samples] Fix TopDownRPG punch animation caused by breaking change in raycast from stride3d#1105

ericwj committed:
Fix debug assertion error in LZ4Stream.Write.

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

ericwj committed:
Typo.

ericwj committed:
Obsolete `SerializationStream.NativeStream` to change the name.

VaclavElias committed:
Readme: Build Instructions improvements (#1494)

ericwj committed:
Comments and whitespace.

palkercode committed:
Added overload for Entity constructor

insomnyawolf committed:
[Debug] DebugTextSystem configurable time on-screen (#1515)

Basewq committed:
[Build] Add build support to referencing projects that use package version with Semantic Versioning 2.0.0 format

Basewq committed:
[Math] Fix Matrix.Invert to return zero matrix if inversion fails instead of NaNs, caused by breaking change from (#1301) (#1520)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

manio143 committed:
Merge pull request #1521 from Basewq/semver-build-support

Add build support to referencing projects that use SemVer 2.0.0 format

palkercode committed:
Merge branch 'stride3d:master' into master

arturoc committed:
openxr: Add static extensionsvariable

To allow to set extensions from outside the engine

Basewq committed:
[Math] Fix Matrix.Decompose, Matrix.DecomposeXYZ & Quaternion.RotationYawPitchRoll to better handle the edge cases where the pitch (or yaw) is +/-90 degrees

Basewq committed:
[Editor] Fix rotation value showing -0 due to MathF respecting negative sign even at zero

manio143 committed:
Configure Samples.Tests and SamplesTestServer to auto resolve packages correctly

manio143 committed:
Merge pull request #1535 from manio143/SamplesTestsAutoResolver

Fix Samples.Tests and SamplesTestServer to auto-resolve packages

Eideren committed:
[Template] Fix first and third person delta time usage (#1511)

Eideren committed:
[Template] FPS, remove unused logic (#1512)

Eideren committed:
Addressing PVS report

Eideren committed:
Strip encoding issues

Eideren committed:
Replace encoding issue

manio143 committed:
Improve documentation of ListFiles and add tests for DatabaseFileProvider implementation

manio143 committed:
Replace string concatenation with interpolation

manio143 committed:
Merge pull request #1547 from manio143/dbFileProviderSearchDocumentation

Improve documentation of ListFiles and add tests for DatabaseFileProvider implementation

arturoc committed:
OpenXR: Fix Non DirectX11 platforms

manio143 committed:
Merge pull request #1469 from ericwj/MigrateInterop2

Improve memory management, embrace modern .NET

Jklawreszuk committed:
Replace native ThreadSleep with .NET Thread.Sleep

manio143 committed:
Merge pull request #1523 from Jklawreszuk/replace-native-sleep-with-net-thread-sleep

Replace native ThreadSleep with .NET Thread.Sleep

Ethereal77 committed:
Use Path.Combine() to concatenate MSBuild paths

manio143 committed:
Merge pull request #1569 from Ethereal77/fix-msbuild-path-combine

Use Path.Combine() to concatenate MSBuild paths

arturoc committed:
Merge branch 'openxr-dx11-master' of github.com:arturoc/stride into openxr-dx11-master

arturoc committed:
VRRendererSettings: Remove CopyMirror property code. Not really useful since this can only be properly evaluated at runtime

arturoc committed:
Update Silk OpenXR nuget

arturoc committed:
VirtualReality project: change OpenXR Silk to 2.10.1 to match other Silk packages in the rest of the engine

manio143 committed:
Merge pull request #1545 from Eideren/addressing_pvs_report

[Misc] Addressing PVS report

dependabot[bot] committed:
Bump System.Drawing.Common from 4.7.0 to 4.7.2 in /sources/engine/Stride

Bumps [System.Drawing.Common](https://github.com/dotnet/corefx) from 4.7.0 to 4.7.2.
- [Release notes](https://github.com/dotnet/corefx/releases)
- [Commits](https://github.com/dotnet/corefx/commits)

---
updated-dependencies:
- dependency-name: System.Drawing.Common
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

manio143 committed:
Merge pull request #1560 from stride3d/dependabot/nuget/sources/engine/Stride/System.Drawing.Common-4.7.2

Bump System.Drawing.Common from 4.7.0 to 4.7.2 in /sources/engine/Stride

palkercode committed:
Merge branch 'stride3d:master' into master

palkercode committed:
Update Entity.cs

Updated with Eideren's suggestion

manio143 committed:
Merge pull request #1499 from palkercode/master

Added overload for Entity constructor

Brian-Hamlin-BCS committed:
Updated Entity.cs

Fixed an error preventing the solution from compiling due to Entity.cs.

manio143 committed:
Merge pull request #1578 from Brian-Hamlin-BCS/master

Updated Entity.cs

skohanowski committed:
Added deconstructors to most of the structs in Stride.Mathematics

skohanowski committed:
Addressed @manio143 feedback regarding deconstructors.

manio143 committed:
Merge pull request #1582 from skohanowski/Deconstructors

Added deconstructors to most of the structs in Stride.Mathematics

mattiascibien committed:
Fix documentation error on Int2

manio143 committed:
Merge pull request #1585 from mattiascibien/fix_int2_docs

Fix documentation error on Int2

Jklawreszuk committed:
Use LZ4 as external dependency

manio143 committed:
Merge pull request #1566 from Jklawreszuk/lz4-package

Use LZ4 as external dependency

Jklawreszuk committed:
Remove settings.json and update gitignore

manio143 committed:
Merge pull request #1586 from Jklawreszuk/vscode-gitignore

Remove settings.json and update gitignore

Eideren committed:
[Rendering] Addressing PVS report - SDL (#1546)

Eideren committed:
[Physics] Update WorldMatrix when forcing transformation on PhysicsComponent (#1406)

xen2 committed:
Merge remote-tracking branch 'arturoc/openxr-dx11-master' (closes #1483)

azeno committed:
[Graphics] If possible use flip model for swap chain

https://devblogs.microsoft.com/directx/dxgi-flip-model

JackPilley committed:
Moved UI rendering to its own stage (Level 10)
Disabled UI from being drawn in the editor (Level 10)

JackPilley committed:
Moved UI to its own render stage (Level 9)
Stopped UI showing in the editor (Level 9)

MechWarrior99 commented:
Hey @JackPilley do you happen to remember what the status of this was? Does some other work need to be done on it before it could be merged? I'd be happy to take a go at it if something is still needed.

JackPilley commented:
> Hey @JackPilley do you happen to remember what the status of this was? Does some other work need to be done on it before it could be merged? I'd be happy to take a go at it if something is still needed.

There's a couple things. I only updated the empty new game project, so I think the samples and templates would also need to be updated. I'm also not sure if the changes would only apply to new projects, or if existing projects will be affected by the changes. And there's always the possibility I did something wrong and it'll break in some situation I didn't think to test.

VaclavElias commented:
Not sure if related. I am adding Clean UI through the code.

In the community toolkit, I am adding default `GraphicsCompositor` like so

`var graphicsCompositor = GraphicsCompositorHelper.CreateDefault(true);`

https://github.com/stride3d/stride-community-toolkit/blob/f774f0bb5a36810e622ceefc5d7ff1dacfd34efb/src/Stride.CommunityToolkit/Engine/GameExtensions.cs#L171-L180

And then, I am applying CleanUI with the extension `AddCleanUIStage()`.

`game.AddGraphicsCompositor().AddCleanUIStage();` 

https://github.com/stride3d/stride-community-toolkit/blob/f774f0bb5a36810e622ceefc5d7ff1dacfd34efb/src/Stride.CommunityToolkit/Rendering/Compositing/GraphicsCompositorExtensions.cs#L37-L43

And making adjustments

https://github.com/stride3d/stride-community-toolkit/blob/f774f0bb5a36810e622ceefc5d7ff1dacfd34efb/src/Stride.CommunityToolkit/Rendering/Compositing/GraphicsCompositorExtensions.cs#L37-L43

```csharp
AddPostEffects(graphicsCompositor);
AddRenderStagesAndFeatures(graphicsCompositor);
```

That means, on the top what is suggested in this issue, not sure if the Stride's ` GraphicsCompositorHelper.CreateDefault(true);` should be also updated so we get the same result, without me having extensions.

Note, that my code implementation adds CleanUI only to `Group31` if I am correct. Not sure if it should be across all groups.

JackPilley commented:
> Note, that my code implementation adds CleanUI only to Group31 if I am correct. Not sure if it should be across all groups.

Since the UI is in its own stage, it shouldn't need to be assigned to any particular group.

MechWarrior99 commented:
So I messed around with it and the setup seems to work fine, couldn't find any issues, so all good there. I was looking at applying these changes to the samples. And it looks like they use the Level10 graphics compositor setup. But not quite sure the best way to go and actually edit them. Can't copy-paste due to the ref ids all being different, and not sure how to open the samples directly if possible even). Any suggestions?

JackPilley commented:
I spent some time today doing some testing. It seems like the changes apply to all of the templates and samples already. So they don't actually need to be updated.

I also tested creating a project on a previous version of Stride, editing the graphics compositor, and then opening the project again after recompiling Stride with these changes. It definitely does pose a risk of breaking existing projects if they've edited their compositor.

MechWarrior99 commented:
>  It seems like the changes apply to all of the templates and samples already

Oh, just tested and you're right! YAY!

> I also tested creating a project on a previous version of Stride, editing the graphics compositor, and then opening the project again after recompiling Stride with these changes. It definitely does pose a risk of breaking existing projects if they've edited their compositor.

Hmm deffinitly shouldn't, it creates the assets in the project and then keeps no reference to the template used.
I tested it my self to be sure and it didn't apply to an existing compositor.
Steps:
1. Create new template project in current version of stride.
2. Open project, open GraphicsCompositor editing window.
3. Close stride, and click the "Save" button when prompted.
4. Open the same project with stride compiled with these changes.
5. See GraphicsCompositor unchanged.

Is it possible something else was going on with your setup?

JackPilley commented:
Here's what happened for me, the modifications I made the compositor are basically the setup for removing post processing from the UI that Hero Crab documented.

I added a new render stage called EXAMPLE

![image](https://github.com/user-attachments/assets/27a11c7a-a50b-4e1b-8a9f-ab505ca2e06a)

I modified the UIRenderFeature so there's groups going to two different stages

![image](https://github.com/user-attachments/assets/7d8ae64e-44e4-4bfc-a39a-20a9b8d2707b)

I added the new render stage to the entry points

![image](https://github.com/user-attachments/assets/23da0533-3459-4d14-8954-8c4c9ad5f1d6)

Now after applying the changes and recompiling the engine I get this:

The UserInterface render stage got added automatically

![image](https://github.com/user-attachments/assets/1210f6bd-f2ee-44e0-b6e3-78f650df3358)

The render feature still sends groups to different stages, but where the first one is going is now different, also the selected groups got reset for the first one

![image](https://github.com/user-attachments/assets/e8e610fd-1332-4785-a0e8-88bfccc7163b)

And there's now an entry point for the UserInterface stage

![image](https://github.com/user-attachments/assets/75b9809b-b6d2-4ccf-a345-df1937277750)

Make sure you clean the solution and fully rebuild the project when you're testing, Visual Studio doesn't always detect the changes in the files I'm modifying.

MechWarrior99 commented:
I first opened and created the project via the stride launcher. And then reopened it from source. Are you able to test in another project or something? And maybe somone esle can as well?

JackPilley commented:
Ok, creating the project from an engine version downloaded through the launcher and then reopening from an engine compiled from source seems to work fine.

It seems weird to me that there would be a difference in behaviour though. Maybe it's because the version from the launcher is a different release number?

JackPilley commented:
It just occurred to me that this might also mean the changes won't apply to the samples and templates.

MechWarrior99 commented:
Well @JackPilley I figured out what is going on... the graphics compositors that are created in new projects have one of the two default graphics compositors as their archetype. That is why they update!

JackPilley commented:
So if I'm understanding correctly, the samples and templates don't need to be updated, but this is a breaking change since existing projects will use the new archetype?

VaclavElias commented:
> Well @JackPilley I figured out what is going on... the graphics compositors that are created in new projects have one of the two default graphics compositors as their archetype. That is why they update!

Is it the one I mentioned in my comment `var graphicsCompositor = GraphicsCompositorHelper.CreateDefault(true);`

If that one is sorted as well, I can remove my extension 🙂.

JackPilley commented:
@VaclavElias No, it looks like the create default function doesn't use the archetype file, so it'll need to be updated as well. Good thing you mentioned it otherwise I wouldn't have known to look for it.
